03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,416 @@
+
+```{r}
+if (!("pheatmap" %in% installed.packages())) {
+  # Install pheatmap
+  install.packages("pheatmap", update = FALSE)
+}
+
+if (!("DESeq2" %in% installed.packages())) {
+  # Install DESeq2
+  BiocManager::install("DESeq2", update = FALSE)
+}
+```
+
+Attach the `pheatmap` and `DESeq2` libraries:
+
+```{r}
+# Attach the `pheatmap` library
+library(pheatmap)
+
+# Attach the `DESeq2` library
+library(DESeq2)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+
+# Set the seed so our results are reproducible:
+set.seed(12345)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata [tab separated values (TSV)](https://en.wikipedia.org/wiki/Tab-separated_values) file and a data TSV file.
+This chunk of code will read in both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  "data", # Replace with path to your metadata file
+  "metadata_SRP070849.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  "data", # Replace with path to your data file
+  "SRP070849.tsv" # Replace with the name of your data file
+)) %>%
+  # Here we are going to store the gene IDs as rownames so that we can have a numeric matrix to perform calculations on later
+  tibble::column_to_rownames("Gene")
+```
+
+Let's take a look at the metadata object that we read into the R environment.
+
+```{r}
+head(metadata)
+```
+
+Now let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>% dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+Now we are going to use a combination of functions from the `DESeq2` and `pheatmap` packages to look at how are samples and genes are clustering.
+
+## Create a DESeqDataset
+
+We will be using the `DESeq2` package for normalization, which requires us to format our data into a `DESeqDataSet` object.
+We turn the data frame (or matrix) into a [`DESeqDataSet` object](TODO: Add link to header section) and specify which variable labels our experimental groups using the `design` argument.
+From the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/28193779/), we know that the mice with `IDH2` mutant AML were treated with vehicle or AG-221 (the first small molecule in vivo inhibitor of IDH2 to enter clinical trials) and the mice with `TET2` mutant AML were treated with vehicle or 5-Azacytidine (Decitabine, hypomethylating agent).[@AML]
+In this chunk of code, we will specify the `refinebio_treatment` variable to label our experimental groups as it contains the data on the specific treatment given for each sample as a part of the experimental design briefly described above.
```

Comment:
Then we can replace this sentence with something that maybe links out some relevant information about specifying models for DESeq2 and that we are using 1 because we aren't doing any DE test here.

Reply:
Gotcha!